hecate_llvm_gen/src/lib.rs

```diff
@@ -0,0 +1,330 @@
+        let print_int = unsafe { LLVMAddFunction(self.module, c_str_ptr!("print_int"), print_int_fn_type) };
+        let entry = unsafe { LLVMAppendBasicBlock(print_int, c_str_ptr!("entry")) };
+        unsafe { LLVMPositionBuilderAtEnd(self.builder, entry); }
+        let value = unsafe { LLVMGetParam(print_int, 0) };
+        let fmt_str = unsafe { LLVMBuildGlobalStringPtr(self.builder, c_str_ptr!("debug: %d"), c_str_ptr!("print_int_fmt_str")) };
+        unsafe { LLVMBuildCall2(self.builder, printf_fn_type, printf, [ fmt_str, value ].as_mut_ptr(), 2, c_str_ptr!()); }
+        unsafe { LLVMBuildRetVoid(self.builder); }
+        let mut id = None;
+        for (k, v) in references {
+            if *v == "print_int" {
+                id = Some(k)
+            }
+        }
+        let id = id.unwrap();
+        self.llvm_functions.insert(*id, LLVMFunction { 
+            llvm_func: print_int, 
+            llvm_ty: print_int_fn_type 
+        });
+    }
+
+    fn register_function(&mut self, func: &IRFunction, references: &HashMap<RefId<ResolvedRef>, &str>) {
+        let llvm_ty = unsafe {
+            core::LLVMFunctionType(LLVMVoidType(), &mut [].as_mut_ptr(), 0, false as LLVMBool)
+        };
+        let llvm_func = unsafe { LLVMAddFunction(self.module, c_str_ptr!(references[&func.name]), llvm_ty) };
+
+        self.functions.insert(func.name, FunctionBuilder {
+            blocks: HashMap::default(),
+            values: HashMap::default()
+        });
+
+        self.llvm_functions.insert(func.name, LLVMFunction { 
+            llvm_func, 
+            llvm_ty
+        });
+    }
+
+    fn build_function(&mut self, func: &IRFunction) {
+        let function = self.functions.get_mut(&func.name).unwrap();
+        let llvm_func = self.llvm_functions[&func.name].llvm_func;
+        
+        // register labels
+        for instr in &func.instrs {
+            if let IRInstr::Block(b) = instr {
+                unsafe {
+                    let block = core::LLVMAppendBasicBlock(llvm_func, c_str_ptr!());
+                    function.blocks.insert(*b, block);
+                }
+            }
+        }
+
+        for instr in &func.instrs {
+            match instr {
+                IRInstr::Block(b) => function.build_block(self.builder, b),
+                IRInstr::Goto(b) => function.build_goto(self.builder, b),
+                IRInstr::Phi(_, _) => (),
+                IRInstr::Branch(v, then_block, else_block) => function.build_branch(self.builder, v, then_block, else_block),
+                IRInstr::Call(r, f, args) => function.build_call(self.builder, r, f, args,  &self.llvm_functions[f]),
+                IRInstr::Return(v) => function.build_return(self.builder, v),
+                IRInstr::Alloca(r) => function.build_alloca(self.builder, r),
+                IRInstr::Store(p, v) => function.build_store(self.builder, p, v),
+                IRInstr::Load(_, _) => todo!(),
```

Comment:
Is this intended? Maybe one should remove the todo before merging on main. If that is not possible, please supply a comment why the feature can currently not be implemented.